lib/key.h

```diff
@@ -4,13 +4,15 @@
 
 #include <cstdint>
 #include <vector>
+#include <optional>
```

Comment:
this should be removed, curious why did you decide not to use it?

Reply:
O oops I was trying something but it was a c++17 feature